### ISSUE TITLE
Problem authenticating to OC.de

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#7115]
##### Detailed steps causing the problem:
- Open your default web browser and manually login to opencaching.de, select to keep logged in
- Go to Menu - Settings - Services - Opencaching.de
- Start authorization process
- Browser is only opened for a split second (which is normal, as the browser login is still valid)
- c:geo reports successful outcome "c:geo is now authorized..."

##### Actual behavior after performing these steps:
When returning to the main menu, it shows OC.DE - Not authorized

##### Expected behavior after performing these steps:
User should be authorized to OC.DE

##### Version of c:geo used:
2018.08.14 and 2018.08.14-NB1

##### Is the problem reproducible for you?
Yes

The only workaround is to manually log out on your default browser and only then start the authentication procedure in c:geo. This will open the web page, let you log in and authorization is working normal afterwards

@following5 @mambofive
Any server side changes recently, which could cause this?

**Comment by following5** [comment]
mambofive and me don't have access to OC.de servers. Server admins are @teiling88 and @bohrsty.

**Comment by Lineflyer** [comment]
For some reason it looks like we send the token twice, which leads to 401 for the second try of course as it was already used:
```
08-15 10:25:59.933 28611 28611 I cgeo    : [main] Requesting settings backup with settings manager
08-15 10:26:01.356 28611  1463 D cgeo    : [OkHttp] GET https://www.opencaching.de/okapi/services/oauth/request_token?oauth_callback=callback%3A%2F%2Fwww.cgeo.org%2Fopencaching.de%2F&oauth_consumer_key=REMOVED3&oauth_nonce=REMOVED1&oauth_signature_method=HMAC-SHA1&oauth_timestamp=1534321561&oauth_token=&oauth_version=1.0&oauth_signature=REMOVED2
08-15 10:26:01.815 28611  1463 D cgeo    : [OkHttp] 200 (459 ms) GET https://www.opencaching.de/okapi/services/oauth/request_token?oauth_callback=callback%3A%2F%2Fwww.cgeo.org%2Fopencaching.de%2F&oauth_consumer_key=REMOVED3&oauth_nonce=REMOVED1&oauth_signature_method=HMAC-SHA1&oauth_timestamp=1534321561&oauth_token=&oauth_version=1.0&oauth_signature=REMOVED2 (http/1.1)
08-15 10:26:02.916 28611  1463 D cgeo    : [OkHttp] POST https://www.opencaching.de/okapi/services/oauth/access_token
08-15 10:26:02.988 28611  1511 D cgeo    : [OkHttp] POST https://www.opencaching.de/okapi/services/oauth/access_token
08-15 10:26:03.087 28611  1463 D cgeo    : [OkHttp] 200 (171 ms) POST https://www.opencaching.de/okapi/services/oauth/access_token (http/1.1)
08-15 10:26:04.438 28611  1511 D cgeo    : [OkHttp] 401 [Unauthorized] (1450 ms) POST https://www.opencaching.de/okapi/services/oauth/access_token (http/1.0)

08-15 10:26:05.859 28611 28611 I cgeo    : [main] Requesting settings backup with settings manager
08-15 10:26:05.896 28611 28611 D cgeo    : [main] LocationProvider: requesting most precise locations

**Comment by teiling88** [comment]
> Any server side changes recently, which could cause this?

No, the last deployment was a bugfix for the email_problems.php file.

**Comment by Lineflyer** [comment]
I had the chance to test this on two other devices with the result, that it is working normal on those.

Seems to be device or Android version specific, but I have no idea what could cause this.

**Comment by Lineflyer** [comment]
Addition:
It seems to be some timing related issue:
- Normally I use Chrome as default browser which is called immediately when starting authorization
- When I clear the defaults, so that need to manually select Chrome as target app when starting autohrization, the problem does not appear

**Comment by okainov** [comment]
BTW this should be fixed with #7170, in particular commit 198f467
